### ISSUE TITLE
GenCast

### ISSUE THREAD

**Issue by dkokron** [NOAA-EMC/graphcast#59]
Does NOAA have any efforts to investigate the DeepMind GenCast model?

**Comment by LinlinCui-NOAA** [comment]
@dkokron Thanks for creating this issue. Yes, we are currently evaluating the GenCast with GDAS data.

**Comment by dkokron** [comment]
I too am doing the same.  I have training an inferencing working at 1 degree.  The results I'm getting are less than hoped for.  It is quite likely that I am doing something incorrectly.  Are you willing to share your experience?

**Comment by LinlinCui-NOAA** [comment]
@dkokron The big difference is SST. There is no SST output in GDAS data, so we use surface temperature by masking out land points.   In addition, we found out that temperature over ice points is using sea ice temp which is too low compared with SST in era5. So we replace SST over ice points with a fixed value (269.9 K).

**Comment by dkokron** [comment]
Yes, I'm doing the same thing as a proxy for SST.
```
# Convert surface temperature into sea surface temperature using land_sea_mask to mask off land values
sst = batch['sea_surface_temperature']
lsm = batch['land_sea_mask']
batch['sea_surface_temperature'][:,:,:,:] = xarray.where(lsm < 0.1, sst[:,:,:,:], np.nan)
```
There is a big difference between that and the sst from ERA5 toward the poles.  ERA5 sst (which is based on OSTIA) never goes below ~0C while the proxy goes way below 0C presumably because it is really the ice temperature.

Are you seeing reasonable predictions from your trained model?  How many epochs did you need to get reasonable results?

**Comment by LinlinCui-NOAA** [comment]
We did not train it from scratch but tried to finetune GenCast with 0.25 degree data but even with H100 still facing OOM issues. So I cannot answer your question. I am curious how worse is your trained model?

**Comment by dkokron** [comment]
![Image](https://github.com/user-attachments/assets/f0c4ef55-5b88-4936-8aed-49505817dfda)

This is 12 hour prediction using the "GenCast 1p0deg <2019.npz" model from DeepMind and GDAS inputs from 2025-02-01

**Comment by dkokron** [comment]
![Image](https://github.com/user-attachments/assets/76b86e9a-ad81-4a8a-aa2e-170c9088e6e3)

Same input but using my trained model.  Trained with a single pass though GDAS data from 31 March 2021 through 1 Jan. 2025 (6 hour increments).  I also did not train from scratch but "refined" from "GenCast 1p0deg <2019.npz"

**Comment by dkokron** [comment]
I am in the middle of figuring out how to swap out my SST proxy with the SST from ERA5.

**Comment by LinlinCui-NOAA** [comment]
Thanks for sharing. How many epochs did you train the model? It looks to me that it needs more epochs to denoise.

**Comment by dkokron** [comment]
I used a single epoch.  I understand DeepMind used way more epochs (https://github.com/google-deepmind/graphcast/issues/136).  They could not tell me how many epochs were actually needed to get reasonable results.

Keep in mind that they trained their model at 1x1 degree for 1200 epochs before switching to 0.25 degree for 38 epochs.  I recommend you do the same.  That should prevent the OOM (at least while running 1x1 degree).
